crypto/evp/p_lib.c

```diff
@@ -934,12 +945,10 @@ void *evp_pkey_make_provided(EVP_PKEY *pk, OPENSSL_CTX *libctx,
 
 #ifndef FIPS_MODE
     /*
-     * If there is an underlying legacy key and it has changed, invalidate
-     * the cache of provider keys.
+     * If there is a legacy "origin" key and it has changed, invalidate
+     * the operation cache of provider keys.
      */
     if (pk->pkey.ptr != NULL) {
-        EVP_KEYMGMT *legacy_keymgmt = NULL;
-
         /*
          * If there is no dirty counter, this key can't be used with
          * providers.
```

Comment:
could ameth be null?

Reply:
I've redone this part quite a bit, please look again